File: packages/nlu/kotlin/nlukit/src/main/kotlin/com/bridgething/nlukit/NluInference.kt
package com.bridgething.nlukit

import com.bridgething.schema.NluSlots

data class NluInferenceOutput(
    val intentLogits: List<Double>,
    val inDomainLogit: Double,
    val slots: NluSlots = NluSlots(),
)

interface NluInferring {
    suspend fun infer(transcript: String): NluInferenceOutput
}

interface NluPrewarmable {
    suspend fun prewarm()
}

data class NluRejectionPolicy(
    val inDomainThreshold: Double = 0.5,
    val clarifyMargin: Double = 0.15,
    val maxAlternates: Int = 2,
)

File: packages/companion/swift/Sources/BridgethingCompanion/NluInference.swift
import Foundation

public struct NluInferenceOutput: Sendable {
    public let intentLogits: [Double]
    public let inDomainLogit: Double
    public let slots: NluMutableSlots

    public init(intentLogits: [Double], inDomainLogit: Double, slots: NluMutableSlots = .init()) {
        self.intentLogits = intentLogits
        self.inDomainLogit = inDomainLogit
        self.slots = slots
    }
}

public protocol NluInferring: Sendable {
    func infer(transcript: String) async throws -> NluInferenceOutput
}

public protocol NluPrewarmable: Sendable {
    func prewarm() async
}
